src/provider/mod.rs

```diff
@@ -59,13 +324,193 @@ where
         self.client().request_noparams("zks_L1BatchNumber").into()
     }
 
-    /// Estimate transaction gas for EIP712 transactions.
+    /// Estimates transaction gas for EIP712 transactions.
     fn estimate_fee(
         &self,
         tx: TransactionRequest,
     ) -> ProviderCall<T, (TransactionRequest,), Eip712Fee> {
         self.client().request("zks_estimateFee", (tx,)).into()
     }
+
+    /// Estimates the gas required for an L1 to L2 transaction.
+    fn estimate_gas_l1_to_l2(
+        &self,
+        tx: TransactionRequest,
+    ) -> ProviderCall<T, (TransactionRequest,), U256> {
+        self.client().request("zks_estimateGasL1ToL2", (tx,)).into()
+    }
+
+    /// Retrieves the bridge hub contract address.
+    fn get_bridgehub_contract(&self) -> ProviderCall<T, NoParams, Option<Address>> {
+        self.client()
+            .request_noparams("zks_getBridgehubContract")
+            .into()
+    }
+
+    /// Retrieves the addresses of canonical bridge contracts for ZKsync Era.
+    fn get_bridge_contracts(&self) -> ProviderCall<T, NoParams, BridgeAddresses> {
+        self.client()
+            .request_noparams("zks_getBridgeContracts")
+            .into()
+    }
+
+    /// Retrieves the L1 base token address.
+    fn get_base_token_l1_address(&self) -> ProviderCall<T, NoParams, Address> {
+        self.client()
+            .request_noparams("zks_getBaseTokenL1Address")
+            .into()
+    }
+
+    /// Lists confirmed tokens.
+    /// Confirmed in the method name means any token bridged to ZKsync Era via the official bridge.
+    ///
+    /// The tokens are returned in alphabetical order by their symbol.
+    /// This means the token id is its position in an alphabetically sorted array of tokens.
+    fn get_confirmed_tokens(&self, from: u32, limit: u8) -> ProviderCall<T, (u32, u8), Vec<Token>> {
+        self.client()
+            .request("zks_getConfirmedTokens", (from, limit))
+            .into()
+    }
```

Comment:
This method is deprecated and will be removed in the future, no need to add it.

Reply:
Deleted.